### DISCUSSION TITLE
Active link highlighting?

### DISCUSSION THREAD

**Discussion by blitzbohne** [TanStack/router#911] [Q&A]
Hi,

I have a link with goes to the root "/" so I set activeProps={{exact:true}} to ensure it doesn't hit any other subroutes. However I also support subroutes on my root like "/dashboard" and of course the link doesn't highlight then. What's the proper solution to handle those special cases?

thanks
Alex

**Comment by StephenSHorton** [comment]
This may not be the answer, but you *can* determine active link state using your own expression.
```tsx
const { location: { pathname } } = useRouterState()
const isActive = href.split('/')[1] === pathname.split('/')[1] // example matches '/foo' | '/foo/bar | '/foo/bar/1'
```

**Comment by ashishsurya** [comment]
@blitzbohne `tanstack-router` attaches a class `active` to all of the `a` tags if they are active, you can use that class as a selector and style your way

**Comment by WinglessFrame** [comment]
As @ashishsurya replied, the active class is applied. For those using Tailwind, you can use the `[&.active]:` className for styling. For example:
For example
```TSX
className="[&.active]:text-green-500"
```

Also, you might face an issue where all matched routes are highlighted. In this case, you should pass the activeOptions={{ exact: true }} prop.

```TSX
<Link
  className="[&.active]:text-green-500 flex items-center gap-1"
  activeOptions={{ exact: true }}
>
  Your link
</Link>
```

**Comment by Code-Victor** [comment]
I have a component like this:
```tsx
            <Link
              to="/"
              activeProps={{
                className: "text-accent-foreground bg-primary/10",
              }}
              search={{ date: format(searchDate, "dd-MM-yyyy") }}
            >
              <Pill />
              <span>View Medicines</span>
          </Link>
```

The link is highlighted when I have a URL like `http://localhost:5173/?date=04-01-2025` but isn't highlighted when I have `http://localhost:5173`. How can I get it to be highlighted, too, please?

**Reply by maurer2** [reply]
Hello, I believe `activeOptions={{ includeSearch: false }}` needs to be set, so that the search params are ignored.
